### PR TITLE
Remove Pragma Once

### PR DESCRIPTION
**Pull request by hamptonm1**
Removing Pragma Once and replacing it with include guards to follow IBM set precedent which can be found here:
https://www.ibm.com/docs/en/zos/2.1.0?topic=messages-ccn8735

```
GCC supports several types of pragmas, primarily in order to compile code originally written for other compilers. Note that in general we do not recommend the use of pragmas; See [Declaring Attributes of Functions](https://gcc.gnu.org/onlinedocs/gcc/extensions-to-the-c-language-family/declaring-attributes-of-functions.html#function-attributes), for further explanation.

The GNU C preprocessor recognizes several pragmas in addition to the compiler pragmas documented here. Refer to the CPP manual for more information.
```

https://gcc.gnu.org/onlinedocs/gcc/extensions-to-the-c-language-family/pragmas-accepted-by-gcc.html#

### CODE REVIEW

File: src/Dialect/ONNX/ONNXOps/OpHelper.hpp [hunk 1]
```diff
@@ -4,15 +4,16 @@
 
 //===------- ONNXOpsHelper.hpp - Helper functions for ONNX dialects -------===//
```
Comment (mikeessen) [inline src/Dialect/ONNX/ONNXOps/OpHelper.hpp:5]: This is a nit, but the prolog "ONNXOpsHelper.hpp" does not match the file name "OpHelper.hpp"

Comment (hamptonm1) [reply src/Dialect/ONNX/ONNXOps/OpHelper.hpp:5]: Thanks I agree, that was there prior. Some files need to be cleaned up.

File: src/Compiler/DisposableGarbageCollector.hpp [hunk 2]
```diff
@@ -8,7 +8,8 @@
 //
 //===----------------------------------------------------------------------===//
 
-#pragma once
+#ifndef ONNX_MLIR_GARABAGE_COLLECTOR_H
```
Comment (mikeessen) [inline src/Compiler/DisposableGarbageCollector.hpp:11]: GARABAGE misspelled

Comment (hamptonm1) [reply src/Compiler/DisposableGarbageCollector.hpp:11]: Thanks! Will fix

**Review by mikeessen — APPROVED** [review]
Approved with minor comments.  Thanks!

File: src/Compiler/DisposableGarbageCollector.hpp (hunk 2 above)
Comment (hamptonm1) [reply src/Compiler/DisposableGarbageCollector.hpp]: Fixed

**Comment by AlexandreEichenberger** [comment]
@hamptonm1 thanks for the upgrade, much appreciated

**Pull request merged**